lib/archethic_web/live/chains/reward_live.ex

```diff
@@ -132,18 +132,24 @@ defmodule ArchethicWeb.RewardChainLive do
     {nb_drops, display_limit} =
       if nb_drops < 0, do: {0, @display_limit + nb_drops}, else: {nb_drops, @display_limit}
 
-    Reward.genesis_address()
-    |> TransactionChain.list_chain_addresses()
-    |> Stream.drop(nb_drops)
-    |> Stream.take(display_limit)
-    |> Stream.map(fn {addr, timestamp} ->
-      %{
-        address: addr,
-        type: (TransactionChain.get_transaction(addr, [:type]) |> elem(1)).type,
-        timestamp: DateTime.from_unix(timestamp, :millisecond) |> elem(1)
-      }
-    end)
-    |> Enum.reverse()
+    case Reward.genesis_address() do
+      address when is_binary(address) ->
+        address
+        |> TransactionChain.list_chain_addresses()
+        |> Stream.drop(nb_drops)
+        |> Stream.take(display_limit)
+        |> Stream.map(fn {addr, timestamp} ->
+          %{
+            address: addr,
+            type: (TransactionChain.get_transaction(addr, [:type]) |> elem(1)).type,
+            timestamp: DateTime.from_unix(timestamp, :millisecond) |> elem(1)
+          }
+        end)
+        |> Enum.reverse()
+
+      _ ->
+        [%{address: "", type: "", timestamp: ""}]
```

Comment:
Why are you not returning an empty list?
What is the unmatched case ? nil ?

Reply:
nice catch. True its nil

Reply:
Ok but why are you not displaying an empty list?